torchgen/gen.py

```diff
@@ -174,6 +176,16 @@ def parse_native_yaml_struct(
             # Only cuda-like devices in tree require device guards
             device_guard=is_cuda_dispatch_key(k),
             index=v,
+            # Which dispatch keys natively support symint
+            # Note: DispatchKey.CompositeExplicitAutograd has to match out
+            # composites; I think there's some factoring problem here
+            symint=k
+            in [
+                DispatchKey.Meta,
+                DispatchKey.CompositeImplicitAutograd,
+                DispatchKey.CompositeExplicitAutograd,
+                DispatchKey.CompositeExplicitAutogradNonFunctional,
```

Comment:
What makes this confusing is that you can have a bundled entry like this:

```
CPU, Meta: foo
```

and when you SymInt'ify the function, this will actually create TWO overloads (one for CPU, one for Meta) but this won't be obvious to the user until the linker error happens.

Reply:
hmmm yeah. I guess we could add a lint for that in the codegen (make it an error in the yaml to have two dispatch entries with the same kernel name, if one dispatch key is in this list and one isn't). If you don't wanna bother with that for now though n